llvm-spirv/lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -288,6 +288,49 @@ void SPIRVRegularizeLLVMBase::expandVIDWithSYCLTypeByValComp(Function *F) {
       nullptr, &Attrs, true);
 }
 
+// intel/llvm customization
+void SPIRVRegularizeLLVMBase::finishSROACooperativeMatrix(Module *M) {
+  for (auto &F : *M) {
+    if (!F.getName().starts_with("_Z19__spirv_AccessChain"))
+      continue;
+    for (auto I = F.user_begin(), E = F.user_end(); I != E;) {
+      if (auto *CI = dyn_cast<CallInst>(*I++)) {
```

Comment:
```suggestion
      auto *CI = dyn_cast<CallInst>(*I++));
      if (!CI)
        continue;
```

You have a pretty huge `if` here, early exit should simplify it